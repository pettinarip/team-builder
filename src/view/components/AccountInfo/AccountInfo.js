import React from 'react'
import PropTypes from 'prop-types'

const AccountInfo = ({ user, authenticated, signOut }) => (
  <section className='account'>
    <div className='account__name'>Hello, {user ? user.displayName : 'mysterious user'}</div>
    <div className='account__options'>
      {
        authenticated &&
        <a href='#' className='account__options__signout button' onClick={signOut}>Sign out</a>
      }
    </div>
  </section>
)

AccountInfo.propTypes = {
  user: PropTypes.object,
  authenticated: PropTypes.bool.isRequired,
  signOut: PropTypes.func.isRequired
}

export default AccountInfo
