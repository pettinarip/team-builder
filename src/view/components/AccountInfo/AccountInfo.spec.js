import React from 'react'
import { shallow } from 'enzyme'
import AccountInfo from './AccountInfo'

describe('AccountInfo component', () => {
  it('should render the component properly', () => {
    const component = mountAccountInfo()
    expect(component).toMatchSnapshot()
  })

  it('should show the user name', () => {
    const component = mountAccountInfo({
      name: 'Foo'
    })
    expect(component).toMatchSnapshot()
  })

  it('should render the sign out button if it is authenticated', () => {
    const component = mountAccountInfo({
      authenticated: true
    })
    expect(component).toMatchSnapshot()
  })

  it('should call sign out function when sign out button is clicked', () => {
    const signOutFn = jest.fn()
    const component = mountAccountInfo({
      authenticated: true,
      signOut: signOutFn
    })

    component.find('.button').simulate('click')
    expect(signOutFn).toHaveBeenCalled()
  })
})

function mountAccountInfo (props) {
  const propsToUse = {
    signOut () {},
    authenticated: false,
    ...props
  }

  return shallow(<AccountInfo {...propsToUse} />)
}
