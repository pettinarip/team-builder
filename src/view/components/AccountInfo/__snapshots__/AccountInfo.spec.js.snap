// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountInfo component should render the component properly 1`] = `
<section
  className="account"
>
  <div
    className="account__name"
  >
    Hi, 
    mysterious user
  </div>
  <div
    className="account__options"
  />
</section>
`;

exports[`AccountInfo component should render the sign out button if it is authenticated 1`] = `
<section
  className="account"
>
  <div
    className="account__name"
  >
    Hi, 
    mysterious user
  </div>
  <div
    className="account__options"
  >
    <button
      className="account__options__signout button"
      onClick={[Function]}
      type="button"
    >
      Sign out
    </button>
  </div>
</section>
`;

exports[`AccountInfo component should show the user name 1`] = `
<section
  className="account"
>
  <div
    className="account__name"
  >
    Hi, 
    Foo
  </div>
  <div
    className="account__options"
  />
</section>
`;
